hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -426,12 +426,33 @@ void GenericSystem::initialize_storage_vectors(
   for (auto i = 0u; i < info_.joints.size(); i++)
   {
     const auto & joint = info_.joints[i];
-    for (auto j = 0u; j < interfaces.size(); j++)
+    for (const auto & interface : joint.state_interfaces)
     {
-      auto it = joint.parameters.find("initial_" + interfaces[j]);
-      if (it != joint.parameters.end())
+      auto it = std::find(interfaces.begin(), interfaces.end(), interface.name);
+
+      // If interface name is found in the interfaces list
+      if (it != interfaces.end())
       {
-        states[j][i] = std::stod(it->second);
+        auto index = std::distance(interfaces.begin(), it);
+
+        // Check the initial_value param is used
+        if (interface.initial_value != "")
```

Comment:
Would also this work?

```suggestion
        if (!interface.initial_value.empty())
```